src/components/CreateGraph/CreateGraph.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect } from 'react';
+import { PlusCircleFilled, MinusCircleFilled, DeleteFilled } from '@ant-design/icons';
+import { Button, Flex, message } from 'antd';
+import Graph from '../Graph/Graph';
+
+const CreateGraph = () => {
+    const [nodes, setNodes] = useState([]);
+    const [edges, setEdges] = useState([]);
+
+    useEffect(() => {
+        const storedNodes = JSON.parse(localStorage.getItem('graphNodes')) || [];
+        const storedEdges = JSON.parse(localStorage.getItem('graphEdges')) || [];
+
+        if (storedNodes.length > 0 && storedEdges.length > 0) {
+            setNodes(storedNodes);
+            setEdges(storedEdges);
+        }
+    }, []);
+
+    useEffect(() => {
+        localStorage.setItem('graphNodes', JSON.stringify(nodes));
+        localStorage.setItem('graphEdges', JSON.stringify(edges));
+    }, [nodes, edges]);
+
+    const addNode = () => {
+        const newNode = { x: Math.random() * 300, y: Math.random() * 300 };
+        const nodeExists = nodes.some(node => node.x === newNode.x && node.y === newNode.y);
+        if (nodeExists) {
+            message.error('Node already exists.');
+            return;
```

Comment:
Extract hardcoded strings into constants file (talk to noa about this she know the structor for this)